app/models/concerns/sluggable.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Sluggable
+  extend ActiveSupport::Concern
+
+  included do |base|
+    base.validates :slug, presence: true, uniqueness: true, format: /\A[a-z0-9\-_]+\z/
+    base.before_validation :generate_slug
+  end
+
+  def generate_slug
+    return true if slug.present?
+    attrs = Array slug_attrs
+    slug = attrs.shift.to_s.parameterize
```

Comment:
Picks the first value from the attributes... this is prepared for a scenario where in a class we may specify `[title, created_at.month, created_at.year]` for example. 

So it's not going to use all of the attributes... just the first if there is none other object with the same title! But if there is.. it will try to create a title-month and so on